tutorials/klusterai-api/integrating-tools.ipynb

```diff
@@ -0,0 +1,934 @@
+    "\n",
+    "By registering several tools in a single `tools` array, we give the LLM a menu of options. We’ll demonstrate with the question:\n",
+    "\n",
+    "> “If Earth’s temperature rises by 2 °C, what percentage increase is that from the current average of 15 °C?”\n",
+    "\n",
+    "To answer, the model only needs arithmetic, but if the prompt also required live data—e.g. “*…and cite a recent study on global warming*”—it could call **two** tools in one turn: first `web_search`, then `calculator`. Here’s how we orchestrate that multi-tool workflow:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. **Describe each tool** – supply JSON-schema specs for `calculator` and `web_search`.\n",
+    "2. **Let the LLM plan** – pass both specs in `multi_tools`; the LLM can issue one or many `tool_calls`.\n",
+    "3. **Dispatch and execute** – `process_multi_tool_calls()` loops over each call, runs the matching helper, and returns results as `{role:\"tool\"}` messages.\n",
+    "4. **Finish in plain English** – a follow-up `chat.completions.create()` lets the model weave everything into a readable answer."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Describe multiple tools\n",
+    "multi_tools = [\n",
+    "    {\n",
+    "        \"type\": \"function\",\n",
+    "        \"function\": {\n",
+    "            \"name\": \"calculator\",\n",
+    "            \"description\": \"Perform arithmetic calculations\",\n",
+    "            \"parameters\": {\n",
+    "                \"type\": \"object\",\n",
+    "                \"properties\": {\n",
+    "                    \"expression\": {\n",
+    "                        \"type\": \"string\",\n",
+    "                        \"description\": \"The mathematical expression to evaluate\"\n",
+    "                    }\n",
+    "                },\n",
+    "                \"required\": [\"expression\"]\n",
+    "            }\n",
+    "        }\n",
+    "    },\n",
+    "    {\n",
+    "        \"type\": \"function\",\n",
+    "        \"function\": {\n",
+    "            \"name\": \"web_search\",\n",
+    "            \"description\": \"Search the web for information\",\n",
+    "            \"parameters\": {\n",
+    "                \"type\": \"object\",\n",
+    "                \"properties\": {\n",
+    "                    \"query\": {\n",
+    "                        \"type\": \"string\",\n",
+    "                        \"description\": \"The search query\"\n",
+    "                    }\n",
+    "                },\n",
+    "                \"required\": [\"query\"]\n",
+    "            }\n",
+    "        }\n",
+    "    }\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Dispatcher helper** – run each requested tool, then feed the results back to the model."
```

Comment:
I'm not sure what's going on with this. It's formatted like a list item, but what for?